modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -617,15 +626,33 @@ interface ApiUrlHandler {
  *
  * @example-ref examples/portal/apiUrl.js
  *
- * @param {object} params Input parameters as JSON.
- * @param {string} params.application Application to reference to the API.
- * @param {string} [params.api] Name of the API
- * @param {string} [params.type=server] URL type. Either `server` (server-relative URL) or `absolute` or `websocket`.
- * @param {object} [params.params] Custom parameters to append to the URL.
+ * @param {object} urlParams Input parameters as JSON.
+ * @param {string} urlParams.application Application to reference to the API.
+ * @param {string} [urlParams.api] Name of the API
+ * @param {string} [urlParams.type=server] URL type. Either `server` (server-relative URL) or `absolute` or `websocket`.
+ * @param {string|string[]} [urlParams.path] Path(s) to be appended to the base URL following the api segment to complete request URL.
+ * @param {object} [urlParams.params] Custom parameters to append to the URL.
  *
  * @returns {string} The generated URL.
  */
-export function apiUrl(params: ApiUrlParams): string {
+export function apiUrl(urlParams: ApiUrlParams): string {
+    checkRequired(urlParams, 'application');
+
+    const {
+        application,
+        api,
+        type = 'server',
+        path = [],
+        params,
+    } = urlParams ?? {};
+
     const bean = __.newBean<ApiUrlHandler>('com.enonic.xp.lib.portal.url.ApiUrlHandler');
-    return bean.createUrl(__.toScriptValue(params));
+    bean.setPath(([] as string[]).concat(path));
```

Comment:
Single string must not be interpreted as single element array.
We should reserve string to preencoded path.